src/mysql-util/src/desc.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct MySqlTableDesc {
+    /// In MySQL the schema and database of a table are synonymous.
+    pub schema_name: String,
+    /// The name of the table.
+    pub name: String,
+    pub columns: Vec<MySqlColumnDesc>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct MySqlColumnDesc {
+    /// The name of the column.
+    pub name: String,
+    /// The MySQL datatype of the column.
+    pub data_type: MySqlDataType,
+    /// Whether the column is nullable.
+    pub nullable: bool,
+    /// The position of the column in the table.
+    pub ordinal_position: u16,
```

Comment:
That's a good point, tracking it as the index of the `MySqlColumnDesc` in its container would be more elegant.

My reasoning here was that for the prototype we may skip some columns if they use unsupported data types, and therefore would need to skip over those indexes when decoding rows we receive.

However based on the other feedback it sounds like we should be erroring-on rather than ignoring columns we encounter with unsupported data-types, so I can switch this to use the ordinal_position as the container-order instead.